back-end/routes/clusters.js

```diff
@@ -0,0 +1,208 @@
+const express = require('express');
+const { PrismaClient } = require('@prisma/client');
+const fetch = require('node-fetch');
+const router = express.Router();
+const prisma = new PrismaClient();
+
+router.use(express.json());
+
+const fetchAudioFeatures = async (songs, accessToken) => {
+    const songIds = songs.map(song => (song.uri).split(':').pop());
+    const stringOfIds = songIds.join(',');
+    try {
+        const response = await fetch(`https://api.spotify.com/v1/audio-features?ids=${encodeURIComponent(stringOfIds)}`, {
+            headers: {
+                'Authorization': `Bearer ${accessToken}`
+            }
+        });
+        if (!response.ok) {
+            throw new Error(`HTTP error! status: ${response.status}`);
+        }
+        const audioData = await response.json();
+        return audioData.audio_features;
+    } catch (error) {
+        console.error(error);
+        return [];
+    }
+};
+
+const getAccessToken = async (userId) => {
+    const token = await prisma.token.findFirst({
+        where: {
+            userId
+        },
+        orderBy: {
+            createdAt: 'desc'
+        }
+    });
+    return token ? token.token : null;
+};
+
+function getValenceFeatures(audioFeatures) {
+    const valenceFeatures = [];
+    audioFeatures.forEach(feature => {
+        const songId = feature.uri.split(':').pop();
+        const valenceValue = feature.valence;
+        valenceFeatures.push({ id: songId, valence: valenceValue });
+    });
+    return valenceFeatures;
+}
```

Comment:
Should this function live here? Wouldn't it be better to have a file specific to functions such as `getValenceFeatures() { ... }`, `getDanceabilityFeatures() { ... }`, etc?